embassy-rp/src/watchdog.rs

```diff
@@ -107,4 +107,100 @@ impl Watchdog {
             w.set_trigger(true);
         })
     }
+
+    pub fn set_scratch0(&mut self, value: u32) {
```

Comment:
this should be `get/set_scratch(&mut self, index: usize, ..)` If you could turn that into an array in `rp-pac` too it'd be even more awesome.